src/Transports/MassTransit.EventHubIntegration/EventHubIntegration/Checkpoints/PendingConfirmation.cs

```diff
@@ -15,11 +15,12 @@ public class PendingConfirmation :
         readonly TaskCompletionSource<long> _source;
         ProcessEventArgs _eventArgs;
 
-        public PendingConfirmation(string eventHubName, ProcessEventArgs eventArgs)
+        public PendingConfirmation(string eventHubName, ProcessEventArgs eventArgs, CancellationToken cancellationToken)
         {
             _eventHubName = eventHubName;
             _eventArgs = eventArgs;
             _source = TaskUtil.GetTask<long>();
+            cancellationToken.Register(() => _source.TrySetCanceled(cancellationToken));
```

Comment:
This is super expensive to do it per message, and since you aren't releasing it creates in infinite memory leak.